werkzeug/contrib/lint.py

```diff
@@ -126,15 +131,15 @@ class GuardedIterator(object):
 
     def __init__(self, iterator, headers_set, chunks):
         self._iterator = iterator
-        self._next = iter(iterator).next
+        self._next = iter(iterator).__next__
         self.closed = False
         self.headers_set = headers_set
         self.chunks = chunks
 
     def __iter__(self):
         return self
 
-    def next(self):
+    def __next__(self):
```

Comment:
Likewise this doesn't work under Python 2, so additionally you need to apply `implements_iterator` from `werkzeug._contrib` to the entire class.